universal-login-commons/src/core/utils/sanitizers/asTransferDetails.ts

```diff
@@ -1,10 +1,11 @@
-import {asObject, asString, Sanitizer} from '@restless/sanitizers';
+import {asObject, asString, Sanitizer, asNumber} from '@restless/sanitizers';
 import {TransferDetails} from '../../..';
 import {asGasParameters} from './asGasParameters';
 
 export const asTransferDetails: Sanitizer<TransferDetails> = asObject<TransferDetails>({
   to: asString,
   amount: asString,
   transferToken: asString,
+  decimals: asNumber,
```

Comment:
It looks pretty bad... I don't like the idea that `decimals` is one of the `TransferDetails` property. It should be a token property, so we would have something like this:
```ts 
    interface TransferDetails {
        ...
        token: Token
        ...
    }
    interface Token {
        address: string;
        decimals: number;
    }
```

I think that the perfect idea would be to use CurrencyValue (but first, we would need to add property `decimals` to it). I know that this a big refactoring, but I think it is worth trying.